### DISCUSSION TITLE
How to correctly deploy the application on subdirectory ?

### DISCUSSION THREAD

**Discussion by lhoucinecherif** [antfu/vitesse#226] [Q&A]
Hi,
I cant' find a correct way to deploy the built app on a subdirectory on my prod server ! 
I've added the "base" param in the vite.config.ts file, but I got a not found page (catch all page) instead of the home page!!

Thanks

**Comment by arbassic** [comment]
I got the same problem... Is there any workaround to launch built site as a sub-page?

When `vite.config.js` got:

```
  base: '/subdir/'
```

the site is not working as expected.
Router is not working properly: the [all].ts page is always triggered.

**Reply by wangfengyuan** [reply]
for me, when i add 
`base: '/apps/vip/'` in `vite.config.ts`

after that add the baseRoute config below
```
Pages({
      extensions: ['vue', 'md'],
      // add this below
      pagesDir: [
        { dir: 'src/pages', baseRoute: '/apps/vip' },
      ],
 }),
```
it works for me

**Reply by lixianbin1** [reply]
It doesn't work for me. The webpage deployed on the `github pages` still displays the `Not found` page

**Comment by lixianbin1** [comment]
I try to set `base: './'` in `vite.config.js`, However, this is invalid. When I set `base: '/demo/'`, It works

**Comment by ajnacodes** [comment]
i had the same issue, this solved my problem https://vite-plugin-ssr.com/base-url just add base: "/directory-name/", in vite.config.ts

**Comment by williansmartins** [comment]
I set base to: '' in vite.config.js. It works
<img width="406" alt="image" src="https://github.com/antfu/vitesse/assets/4259203/a690fd0b-606a-4c0d-b334-9ef1f94df18b">
